_shared/local_extensions.py

```diff
@@ -110,11 +113,78 @@ def include_exists(self, context, path):
     def include(self, context, path, indent=0):
         """Return the lines from the project's .cookiecutter/includes/{path} or an empty string."""
         try:
-            lines = self._readlines(context, path)
+            file_ = self._open(context, path)
         except (FileNotFoundError, NotADirectoryError):
             return ""
 
-        return textwrap.indent("".join(lines), " " * indent)
+        return textwrap.indent("".join(file_.readlines()), " " * indent)
+
+    @pass_context
+    def include_json(self, context, path):
+        """Return the project's .cookiecutter/includes/{path} data or None.
+
+        Return the contents of the project's .cookiecutter/includes/{path} JSON
+        file deserialized into a Python object.
+
+        Return None if the project has no .cookiecutter/includes/{path} file.
+        """
+        return json.load(self._open(context, path))
+
+    @pass_context
+    def supervisor_settings(self, context):
```

Comment:
There's two reasons why this Supervisor-specific method has to be implemented in Python:

1. It's a total pain to write this logic (setting default values etc) in Jinja. Ends up really ugly
2. With cookiecutter there's no way to shared template code or macros between files. But this Python function is available to all templates